esphome/components/sun_gtil2/sun_gtil2.h

```diff
@@ -0,0 +1,44 @@
+#pragma once
+
+#include "esphome/core/component.h"
+#include "esphome/components/sensor/sensor.h"
+#include "esphome/components/text_sensor/text_sensor.h"
```

Comment:
These includes, and also any code lines that touch the (text)sensor variables have to be wrapped in these preprocessors in case the user is not using the relevant sensors. 

To test this, try compiling a YAML without any `text_sensor` block in it

```suggestion
#include "esphome/core/defines.h"

#ifdef USE_SENSOR
#include "esphome/components/sensor/sensor.h"
#endif
#ifdef USE_TEXT_SENSOR
#include "esphome/components/text_sensor/text_sensor.h"
#endif
```